### PR TITLE
refactor(amplifier): update to async style

### PR DESCRIPTION
**Pull request by eguajardo**
https://axelarnetwork.atlassian.net/browse/AXE-4714

### CODE REVIEW

File: cosmwasm/deploy-contract.js [hunk 1]
```diff
@@ -29,31 +29,28 @@ const upload = (client, wallet, chainName, config, options) => {
     if (!fetchCodeId && (!reuseCodeId || isNil(contractConfig.codeId))) {
```
Comment (milapsheth) [inline cosmwasm/deploy-contract.js:29]: codeId should be extracted here. And then `contractConfig` should be defined as either the top-level config or chain specific (and initialize the chain specific config to {} if it doesn't exist), so you don't have the `if (chainConfig)` checks everywhere

File: cosmwasm/submit-proposal.js [hunk 2]
```diff
@@ -195,53 +190,58 @@ const main = async (options) => {
...
+        }
+
+        default:
+            throw new Error('Invalid proposal type');
+    }
+
+    saveConfig(config, env);
 };
 
 const programHandler = () => {
```
Comment (milapsheth) [inline cosmwasm/submit-proposal.js:247]: not in this PR, but you can refactor the CLI handler to use commands. So options can be defined only for the cmds that need them, similar to axelard CLI
https://github.com/axelarnetwork/axelar-contract-deployments/blob/main/sui/gateway.js#L343-L358

**Review by milapsheth — APPROVED** [review]

File: cosmwasm/submit-proposal.js (hunk 2 above)
Comment (eguajardo) [reply cosmwasm/submit-proposal.js:247]: yeah, I have been wanting to do that, but haven't had the chance to dig deeper into how sub commands work, will take a look at the sui script

File: cosmwasm/deploy-contract.js [hunk 3]
```diff
@@ -29,31 +29,28 @@ const upload = (client, wallet, chainName, config, options) => {
...
-                printInfo('Expected contract address', address);
-            })
-            .then(() => ({ wallet, client }));
-    }
+        const { address, codeId } = await uploadContract(client, wallet, config, options);
+
+        printInfo('Uploaded contract binary');
+        contractConfig.codeId = codeId;
+
+        if (!address) {
```
[the remark this answers is not in the document]
Comment (milapsheth) [reply cosmwasm/deploy-contract.js:37]: this shouldn't be handled by `uploadContract`. Instead once upload completes, if options.instantiate2 was set, then call a `predictAddress` helper with `options` that can return the predicted address.

File: cosmwasm/deploy-contract.js (hunk 1 above)
Comment (eguajardo) [reply cosmwasm/deploy-contract.js:34]: refactoring contractConfig will be done in a different PR after evaluating the best way to also move codeId and other top level contract configs to the contract chain config

**Review by milapsheth — APPROVED** [review]

**Pull request merged**